storage/src/mmr/journaled.rs

```diff
@@ -58,19 +58,19 @@ pub struct Mmr<E: RStorage + Clock + Metrics, H: CHasher, M: Hasher<H>> {
     pruned_to_pos: u64,
 }
 
-impl<E: RStorage + Clock + Metrics, H: CHasher, M: Hasher<H>> Builder<H, M> for Mmr<E, H, M> {
-    async fn add(&mut self, hasher: &mut M, element: &[u8]) -> Result<u64, Error> {
-        self.add(hasher, element).await
+impl<E: RStorage + Clock + Metrics, H: CHasher> Builder<H> for Mmr<E, H> {
+    async fn add(&mut self, hasher: &mut impl Hasher<H>, element: &[u8]) -> Result<u64, Error> {
+        self.mem_mmr.add(hasher, element).await
```

Comment:
This may have been the result of the "recursive compiler error" (the function was calling itself).